libs/traits/src/lib.rs

```diff
@@ -685,3 +685,46 @@ pub trait CurrencyInspect {
 	/// Checks whether the provided currency is a tranche token.
 	fn is_tranche_token(currency: Self::CurrencyId) -> bool;
 }
+
+pub trait TokenSwaps<Account> {
+	type CurrencyId;
+	type Balance;
+	type OrderId;
+	/// Swap tokens buying a `buy_amount` of `currency_in` using the
+	/// `currency_out` tokens. The implementator of this method should know
+	/// the current market rate between those two currencies.
+	/// `sell_price_limit` defines the lowest price acceptable for
+	/// `currency_in` currency when buying with `currency_out`. This
+	/// protects order placer if market changes unfavourably for swap order.
+	/// Returns the order id created with by this buy order if it could not
+	/// be inmediately and completelly fullfilled. If there was already an
+	/// active order with the same account currencies, the order is
+	/// increased/decreased and the same order id is returned.
```

Comment:
`update_order` allows accounts to update active orders they have, and can also be used when we add partial fulfillment.

Reply:
But given that comment, it seems like `place_order` does the same if the order already exits.

Reply:
Ah, the current `place_order` creates a new order instead of updating the order, the docstring wasn't updated to match the behaviour, but it might be better to have behaviour match docstring.
@wischli would it be better for `place_order` to update an existing order for an account/currency pair?
If that's the case it would make sense to switch order id back to hash, but without nonce--this should be a simple change though.

Reply:
Verified current behaviour is good, and updated docstring to make sure it's accurate.